src/zkl.rs

```diff
@@ -0,0 +1,107 @@
+pub struct Parser {}
+
+impl Parser {
+    pub fn parse(proof_spec: &ProofSpec, witness: &Witness) -> Result<Proof, ParseError> {
+        unimplemented!();
+    }
+}
+
+pub struct ProofSpec {}
+
+pub struct ProofSpecBuilder {}
+
+impl ProofSpecBuilder {
+    pub fn new() -> ProofSpecBuilder {
+        ProofSpecBuilder{}
+    }
+}
+
+pub struct Witness {}
+
+pub struct WitnessBuilder {}
+
+impl WitnessBuilder {
+    pub fn new() -> WitnessBuilder {
+        WitnessBuilder {}
+    }
+}
+
+pub struct Proof {}
+
+impl Proof {
+    pub fn verify(&self, proof_spec: &ProofSpec) -> Result<bool, ParseError> {
+        unimplemented!();
+    }
+}
+
+#[repr(i32)]
+#[derive(Debug, Eq, PartialEq)]
+pub enum ParseError {
+    InvalidProofSpec = 1,
+    InvalidWitness = 2
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Need test for attribute disclosure, where disclosure is done but tthe disclosed value does not match expected (integer value in proof does not match the hash of actual value) and other where the proof verification fails